Bug-1538451 documentation for `commands.openShortcutSettings` (#38234)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/commands/openshortcutsettings/index.md

---
title: openShortcutSettings()
slug: Mozilla/Add-ons/WebExtensions/API/commands/openShortcutSettings
page-type: webextension-api-function
browser-compat: webextensions.api.commands.openShortcutSettings
---

{{AddonSidebar}}

> Opens the Manage Extension Shortcuts page of Manage Your Extensions (`about:addons`) and, if the extension has shortcuts, scrolls to and highlights the extension's shortcut key options.

Reviewer: ```suggestion
Opens the browser UI that allows users to configure keyboard shortcuts for an extension.

In Firefox, this is opens the [Manage Extension Shortcuts](https://support.mozilla.org/en-US/kb/manage-extension-shortcuts-firefox) view.

Chrome does not support this method yet. In Chrome, `chrome://extensions/shortcuts` can be opened via the `tabs.create` method, as an alternative.
```

Author: ```suggestion
Opens the browser UI that enables users to configure an extension's keyboard shortcuts.

In Firefox, this function opens the [Manage Extension Shortcuts](https://support.mozilla.org/en-US/kb/manage-extension-shortcuts-firefox) view.

Chrome doesn't support this function yet. In Chrome, `chrome://extensions/shortcuts` can be opened using the `tabs.create` function.
```

---
title: openShortcutSettings()
slug: Mozilla/Add-ons/WebExtensions/API/commands/openShortcutSettings
page-type: webextension-api-function
browser-compat: webextensions.api.commands.openShortcutSettings
---

{{AddonSidebar}}

Opens the browser UI that enables users to configure an extension's keyboard shortcuts.

In Firefox, this function opens the [Manage Extension Shortcuts](https://support.mozilla.org/en-US/kb/manage-extension-shortcuts-firefox) view.

> Chrome doesn't support this function yet. In Chrome, `chrome://extensions/shortcuts` can be opened using the `tabs.create` function.

Reviewer 2: Just did a quick search and it appears that Safari does not allow users to change keyboard shortcuts for web extensions. (Corroborated in the [`commands` compatibility note for Safari](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/commands)). Should we call this out here?

Author: The API is marked as supported from Safari 14. I'll create an issue and tag our Apple contact as we may need a note in the API overview, if the API doesn't do anything practical on Safari.

This is an asynchronous function that returns a [`Promise`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

## Syntax

```js-nolint
let openedShortcutSettings = browser.commands.openShortcutSettings();
```

### Parameters

None.

### Return value

> A {{JSxRef("Promise")}} fulfilled with no arguments if the Manage Extension Shortcuts page is opened. If the request fails, the promise is rejected with an error message.

Reviewer 2: It may be best to reword this to be more generic. Feel free to reject or adapt this draft suggestion.

```suggestion
A {{JSxRef("Promise")}} fulfilled with no arguments if the extension shortcut management interface opens. If the request fails, the promise is rejected with an error message.
```

Author: Good point, probably prudent, even though it may never have any practical impact.
